src/index.ts

```diff
@@ -1,101 +1,90 @@
+import { gt } from 'semver'
+
+interface DownloadOptions {
+  agent?: Agent
+}
 
 export interface Release {
-  repository: string;
-  package: string;
-  destination: string;
-  version: string;
-  extract: boolean;
+  repository: string
+  package: string
+  destination: string
+  version: string
+  extract: boolean
 }
 
-export async function fetchLatest(
-  release: Release,
-  fetchOptions?: RequestInit
-) {
-  release.version = await resolveRelease(release.repository, fetchOptions);
-  return fetchVersion(release, { agent: fetchOptions && fetchOptions.agent });
+export async function fetchLatest(release: Release, fetchOptions?: RequestInit): Promise<void> {
+  // eslint-disable-next-line no-param-reassign
+  release.version = await resolveRelease(release.repository, fetchOptions)
+  return fetchVersion(release, { agent: fetchOptions && fetchOptions.agent })
 }
 
-export async function fetchVersion(
-  release: Release,
-  { agent }: DownloadOptions = {}
-) {
-  validateRelease(release);
-  await downloadFile(release, { agent });
+export async function fetchVersion(release: Release, { agent }: DownloadOptions = {}): Promise<void> {
+  validateRelease(release)
+  await downloadFile(release, { agent })
 }
 
 export async function updateAvailable(
   repository: string,
   currentVersion: string,
-  fetchOptions?: RequestInit
+  fetchOptions?: RequestInit,
 ): Promise<boolean> {
-  const latestVersion = await resolveRelease(repository, fetchOptions);
-  return newerVersion(latestVersion, currentVersion);
+  const latestVersion = await resolveRelease(repository, fetchOptions)
+  return newerVersion(latestVersion, currentVersion)
 }
 
-async function resolveRelease(
-  repository: string,
-  fetchOptions?: RequestInit
-): Promise<string> {
-  const res = await fetch(
-    `https://api.github.com/repos/${repository}/releases/latest`,
-    fetchOptions
-  );
-  const json = await res.json();
-  if (
-    res.status === 403 &&
-    typeof json.message === 'string' &&
-    json.message.includes('API rate limit exceeded')
-  ) {
-    throw new Error('API rate limit exceeded, please try again later');
+async function resolveRelease(repository: string, fetchOptions?: RequestInit): Promise<string> {
+  const res = await fetch(`https://api.github.com/repos/${repository}/releases/latest`, fetchOptions)
+  const json = await res.json()
+  if (res.status === 403 && typeof json.message === 'string' && json.message.includes('API rate limit exceeded')) {
+    throw new Error('API rate limit exceeded, please try again later')
   }
-  return json.tag_name;
+  return json.tag_name
 }
 
 async function downloadFile(release: Release, { agent }: DownloadOptions) {
-  const url = `https://github.com/${release.repository}/releases/download/${release.version}/${release.package}`;
-  mkdirp.sync(release.destination);
+  const url = `https://github.com/${release.repository}/releases/download/${release.version}/${release.package}`
+  sync(release.destination)
```

Comment:
What are your thoughts on renaming this variable?

Reply:
Great point. Fixed in https://github.com/netlify/gh-release-fetch/pull/24/commits/eee8c12904718531ccbdbee071d0571fc5d16a2e (in a different way).

I'll switch to `make-dir` in another PR.